OpenRA.Mods.Cnc/Traits/TransformsNearResources.cs

```diff
@@ -0,0 +1,75 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Cnc.Traits
+{
+	[Desc("Transforms into another actor when a resource spawns adjacent.")]
+	public class TransformsNearResourcesInfo : ITraitInfo
+	{
+		[FieldLoader.Require]
+		[ActorReference] public readonly string IntoActor = null;
+
+		public readonly CVec Offset = CVec.Zero;
+
+		[Desc("Don't render the make animation.")]
+		public readonly bool SkipMakeAnims = false;
+
+		[FieldLoader.Require]
+		[Desc("Resource type which triggers the transformation.")]
+		public readonly string Type = null;
+
+		public virtual object Create(ActorInitializer init) { return new TransformsNearResources(init.Self, this); }
+	}
+
+	public class TransformsNearResources : ITick
+	{
+		readonly TransformsNearResourcesInfo info;
+		readonly ResourceLayer resourceLayer;
+
+		public TransformsNearResources(Actor self, TransformsNearResourcesInfo info)
+		{
+			this.info = info;
+			resourceLayer = self.World.WorldActor.Trait<ResourceLayer>();
```

Comment:
It wouldn't really make sense to not have a `ResourceLayer`, but do we want a `Requires<ResourceLayer>` here?

Reply:
I think we can't as the `ResourceLayer` is at the `World` and this trait is for regular actors.

Reply:
Ah right, my oversight.